scapy/sendrecv.py

```diff
@@ -853,7 +853,11 @@ def prn(pkt):
 
 @conf.commands.register
 def tshark(*args,**kargs):
-    """Sniff packets and print them calling pkt.show(), a bit like text wireshark"""
-    sniff(prn=lambda x: x.display(),*args,**kargs)
-
-
+    """Sniff packets and print them calling pkt.summary(), a bit like text wireshark"""
+    print("Capturing on '" + str(kargs.get('iface') if 'iface' in kargs else conf.iface) + "'")
+    i = [1]  # This should be a nonlocal variable, using a mutable object for Python 2 compatibility
```

Comment:
You should start at `0` and increase before the `print()` in the callback so that your final value is correct.

Reply:
Correct, though tcpdump (see the screenshot above) starts at 1. I was thinking that to be able to compare their results, we should have copied this behavior.

Should I anyway use 0 ?

Reply:
That's why I say to increment before calling `print()`. The problem is not the value displayed with each packet, but the total value. You should have:
```python
    i = [0]
    def _cb(pkt):
        i[0] += 1
        print("%5d\t%s" % (i[0], pkt.summary()))
    sniff(prn=_cb, store=False, *args, **kargs)
    print("\n%d packet%s captured" % (i[0], 's' if i[0] > 1 else ''))
```
Nb: I used `pkt` instead of `x` for the argument, and '%5d' to mimic tshark output.